ui/src/CurrentUser.ts

```diff
@@ -87,16 +87,22 @@ export class CurrentUser {
             .then((passThrough) => {
                 this.user = passThrough.data;
                 this.loggedIn = true;
-                this.hasNetwork = true;
+                this.connectionErrorMessage = null;
+                this.reconnectTime = 7500;
                 return passThrough;
             })
             .catch((error: AxiosError) => {
                 if (!error || !error.response) {
-                    this.lostNetwork();
+                    this.connectionError('No network connection.');
                     return Promise.reject(error);
                 }
 
-                this.hasNetwork = true;
+                if (error.response.status === 503) {
```

Comment:
503 is only service unavailable, we should include all 500 errors.
```suggestion
                if (error.response.status >= 500) {
```